xray/backends/netcdf3.py

```diff
@@ -40,8 +40,8 @@ def coerce_nc3_dtype(arr):
         # TODO: raise a warning whenever casting the data-type instead?
         cast_arr = arr.astype(new_dtype)
         if ((('int' in dtype or 'U' in dtype) and
-                    not (cast_arr == arr).all())
-                or ('float' in dtype and
+                not (cast_arr == arr).all()) or
```

Comment:
Looks like your linter really doesn't like putting conjugations at the start of lines. Is that in PEP8?